### PR TITLE
Feat/sagas sync

### PR DESCRIPTION
**Pull request by j8seangel**
tl;dr: Using sagas to make our life easier.

The sync process of the alerts is now done by [sagas](https://redux-saga.js.org/) instead of a recursive and buggy function. This library helps us to have a better and easier workflow using redux-offline and a powerful actions handling.

This PR achieves:
- [Niiice](https://github.com/Vizzuality/forest-watcher/commit/6be902c5ad8cbd62255a5f55e4c133ff8a2ab0dd) config redux-offline and sagas, 
- Sync alerts after every areas commit action.
- Create showNotification global helper.
- Show notification when no connection on the pull to refresh.
- Include i18n in sync notifications.
- Alerts and layers flow types.
- Fix production build logging the last actions.
- Fix android build after gradle update.

### CODE REVIEW

File: app/main.js [hunk 1]
```diff
@@ -20,9 +14,21 @@ import { setExceptionHandlers, checkPrevCrashes } from './crashes';
 // Show request in chrome network tool
 // GLOBAL.XMLHttpRequest = GLOBAL.originalXMLHttpRequest || GLOBAL.XMLHttpRequest;
 
+function setCodePush() {
+  const codepushEnable = false; // !__DEV__
```
Comment (sorodrigo) [inline app/main.js]: Dead kitten 🐱

File: app/redux-modules/alerts.js [hunk 2]
```diff
@@ -60,13 +62,13 @@ const initialState = {
...
-  pendingData: {}
+  queue: []
 };
 
-export default function reducer(state = initialState, action) {
+export default function reducer(state: AlertsState = initialState, action: AlertsAction) {
   switch (action.type) {
-    case START_APP: {
-      return { ...state, syncError: false, pendingData: {}, clusters: null };
+    case 'app/START_APP': {
```
Comment (sorodrigo) [inline app/redux-modules/alerts.js]: let's add an issue in PT this needs to go

File: app/redux-modules/alerts.js [hunk 3]
```diff
@@ -212,26 +167,25 @@ export function resetAlertsDb() {
...
-    const datasetSlug = dataset && dataset.slug;
     const canDisplay = state().alerts.canDisplayAlerts;
 
-    const action = { type: SET_ACTIVE_ALERTS, payload: null };
-    if (datasetSlug && canDisplay) {
-      activeCluster.supercluster = memoizedAreaToClusters(area.id, datasetSlug, dataset.startDate);
-      action.payload = `${area.id}_${datasetSlug}_${dataset.startDate}`;
+    const action = ({ type: SET_ACTIVE_ALERTS, payload: null }: { type: string, payload: ?string });
+    if (dataset && canDisplay) {
+      activeCluster.supercluster = memoizedAreaToClusters(area.id, dataset.slug, dataset.startDate);
```
Comment (sorodrigo) [inline app/redux-modules/alerts.js]: Let's clear the cache when the alerts are updated. I just realised this is not the case same as LOGOUT_REQUEST.

File: app/sagas/sync.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
+// @flow
+import type { State } from 'types/store.types';
+
```
Comment (sorodrigo) [inline app/sagas/sync.js]: I think we should rename to alerts. What do you think about starting to use a new notation?

File: app/components/toast-notification/styles.js [hunk 5]
```diff
@@ -34,6 +34,12 @@ export default StyleSheet.create({
   successText: {
     color: Theme.fontColors.white
   },
+  warn: {
```
Comment (sorodrigo) [inline app/components/toast-notification/styles.js]: Lastly, I think this color shouldn't be called `warn`. I think of yellow when I read `warn`, maybe `disabled`?

File: app/main.js (hunk 1 above)
Comment (j8seangel) [reply app/main.js]: I thought it is useful until we have the first release. But is a good catch to remind it!

File: app/redux-modules/alerts.js (hunk 3 above)
Comment (j8seangel) [reply app/redux-modules/alerts.js]: 👀

File: app/sagas/sync.js (hunk 4 above)
Comment (j8seangel) [reply app/sagas/sync.js]: 👍
